torch/distributed/_composable/fsdp/_fsdp_state.py

```diff
@@ -142,6 +142,7 @@ def _init_fqns(self) -> None:
             if module in module_to_fsdp_param_group:
                 module_to_fsdp_param_group[module]._module_fqn = module_name
 
+    @torch.compiler.disable
```

Comment:
The proposed changes will prevent `_pre_forward` and `_post_forward` from being traced during full-graph compile.

Discussed offline and a better solution is likely to disable only if `reshard_after_forward` is int.

Reply:
@yf225 I think one challenge here is that there are legitimate cases where the current graph-break tracing approach errors during tracing of the pre-forward hook, preventing using `torch.compile` at all.

Two cases I have seen for this are (1) DTensor-based 2D parallel and (2) most recently when I have been prototyping fp8 all-gather where the `Float8DynamicLinear.weight` is a separate tensor subclass.

I was thinking it may be worth considering some global config that allows us to switch between the graph-break and full-compile approaches since both may need to co-exist for some time (though let me know if you disagree).

Reply:
Having a global config to switch between graph-break compile and full-graph compile sounds great. I can imagine some region doesn't work with graph-break compile but might be able to work with full-graph compile (and vice versa). We could use an `if` statement on the global config to decide whether to dynamo disable the hooks. Wonder wdyt?

Reply:
Global config and `if` statement sounds good to me!

Reply:
thanks for the discussion. I can add a dynamo config ``trace_fsdp_hooks=True/False``. An example UX would be 

```
with torch._dynamo.config.patch(trace_fsdp_hooks=False):
     model = torch.compile(model)
```